scripts/Muon/GUI/Common/contexts/fitting_contexts/fitting_context.py

```diff
@@ -179,46 +181,75 @@ class FitInformation(object):
+        self.covariance_workspace: StaticWorkspaceWrapper = covariance_workspace
+        self.tf_asymmetry_fit: bool = tf_asymmetry_fit
+
+        self._fit_parameters = FitParameters(self.parameter_workspace, global_parameters)
 
     def __eq__(self, other):
         """Objects are equal if each member is equal to the other"""
-        return self.parameter_workspace_name == other.parameter_workspace_name and \
+        return self.input_workspaces == other.input_workspaces and \
             self.fit_function_name == other.fit_function_name and \
-            self.input_workspaces == other.input_workspaces and \
-            self.output_workspace_names == other.output_workspace_names
+            self.output_workspace_names() == other.output_workspace_names() and \
+            self.parameter_workspace.workspace_name == other.parameter_workspace.workspace_name and \
+            self.covariance_workspace.workspace_name == other.covariance_workspace.workspace_name and \
+            self.tf_asymmetry_fit == other.tf_asymmetry_fit
+
+    def output_workspaces_objects(self) -> list:
+        """Returns a list of output workspaces. Do not try place these objects into the ADS as they are not copied."""
+        return [output_workspace.workspace for output_workspace in self.output_workspaces]
+
+    def output_workspace_names(self) -> list:
+        """Returns a list of the output workspace names."""
+        return [output_workspace.workspace_name for output_workspace in self.output_workspaces]
+
+    def output_group_name(self) -> str:
+        """Returns the name of the WorkspaceGroup that contains the output workspaces from a fit."""
+        _, directory = create_fitted_workspace_name(self.input_workspaces[0], self.fit_function_name)
+        return directory[:-1]
 
     @property
     def parameters(self):
         return self._fit_parameters
 
     @property
     def parameter_workspace_name(self):
-        return self._fit_parameters.parameter_workspace_name
+        return self.parameter_workspace.workspace_name
+
+    def add_copy_to_ads(self) -> None:
+        """Adds the output workspaces for this fit to the ADS. Must be a copy so the fit history remains a history."""
+        workspace_group = WorkspaceGroup()
+        add_ws_to_ads(self.output_group_name(), workspace_group)
+
+        for output_workspace in self.output_workspaces:
+            add_ws_to_ads(output_workspace.workspace_name, output_workspace.workspace_copy())
+            workspace_group.add(output_workspace.workspace_name)
+
+        add_ws_to_ads(self.parameter_workspace.workspace_name, self.parameter_workspace.workspace_copy())
+        workspace_group.add(self.parameter_workspace.workspace_name)
+
+        add_ws_to_ads(self.covariance_workspace.workspace_name, self.covariance_workspace.workspace_copy())
+        workspace_group.add(self.covariance_workspace.workspace_name)
 
     def log_names(self, filter_fn=None):
```

Comment:
can we just have an example of a log in the comment? (i.e. It should accept a log (e.g. xxxx))